cryptographic_estimators/PKEstimator/PKAlgorithms/sbc.py

```diff
@@ -108,8 +108,8 @@ def _compute_time_and_memory(self, parameters: dict, verbose_information=None):
         if w1 > w or w < d or n - w < m - d or (d == 1 and w > n - m):
             return inf, inf
 
-        N_w = log2(binomial(n, w)) + log2((q ** d - 1) ** (w - d)) + gauss_binomial(m, d, q) - gauss_binomial(n, d,
-                                                                                                              q)  # number of expected subcodes
+        N_w = log2(binomial(n, w)) + log2((q ** d - 1)) * (w - d) + gauss_binomial(m, d, q) - gauss_binomial(n, d,
+                                                                                                             q)  # number of expected subcodes
```

Comment:
I see that there are a lot of formulas like this, from my point of view, I think we can extract them into smaller functions with descriptive names. For example

```suggestion
        def get_number_of_subcodes(n,w,q,d,m):
           return log2(binomial(n, w)) + log2((q ** d - 1)) * (w - d) + gauss_binomial(m, d, q) - gauss_binomial(n, d, q)
           
        N_w = get_number_of_subcodes(n, w, q, d, m)
```

If they're only used inside a method I'm ok with defining them inside the method.
What do you guys think?